docs/migration_2_to_3.rst

```diff
@@ -2,14 +2,56 @@
 Migrating to ReFrame 3
 ======================
 
+ReFrame 3 brings substantial changes in its configuration.
+The configuration component was completely revised and rewritten from scratch in order to allow much more flexibility in how the framework's configuration options are handled, as well as to ensure the maintainability of the framework in the future.
 
-Updating your tests
+At the same time, ReFrame 3 deprecates some common pre-2.20 test syntax in favor of the more modern and intuitive pipeline hooks.
+
+This guide details the necessary steps in order to easily migrate to ReFrame 3.
+
+
+Updating Your Site Configuration
+--------------------------------
+
```

Comment:
I need to mention that the default settings file has changed. It's no more `reframe/settings.py`.